### DISCUSSION TITLE
run example code get an error

### DISCUSSION THREAD

**Discussion by wangyao520** [gdsfactory/gdsfactory#1434] [Q&A] [answered]
**Hi I ran this code** 

coupler_lengths = [10, 20, 30, 40, 50, 60, 70, 80]
coupler_gaps = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8]
delta_lengths = [10, 100, 200, 300, 400, 500, 500]

mzi = gf.components.mzi_lattice(
    coupler_lengths=coupler_lengths,
    coupler_gaps=coupler_gaps,
    delta_lengths=delta_lengths,
)

# Add fill
c = gf.Component("component_with_fill")
layers = [(1, 0)]
fill_size = [0.5, 0.5]

c << gf.fill_rectangle(
    mzi,
    fill_size=fill_size,
    fill_layers=layers,
    margin=5,
    fill_densities=[0.8] * len(layers),
    avoid_layers=layers,
)

c << mzi
c.show(show_ports=True)

**but got an error, it says** 

TypeError                                 Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_16468/447142379.py in <module>
     14 fill_size = [0.5, 0.5]
     15 
---> 16 c << gf.fill_rectangle(
     17     mzi,
     18     fill_size=fill_size,

~\anaconda3\lib\site-packages\gdsfactory\cell.py in _cell(*args, **kwargs)
    162             )
    163 
--> 164         component = func(*args, **kwargs)
    165 
    166         # if the component is already in the cache, but under a different alias,

~\anaconda3\lib\site-packages\pydantic\decorator.cp39-win_amd64.pyd in pydantic.decorator.validate_arguments.validate.wrapper_function()

~\anaconda3\lib\site-packages\pydantic\decorator.cp39-win_amd64.pyd in pydantic.decorator.ValidatedFunction.call()

~\anaconda3\lib\site-packages\pydantic\decorator.cp39-win_amd64.pyd in pydantic.decorator.ValidatedFunction.execute()

~\anaconda3\lib\site-packages\gdsfactory\fill.py in fill_rectangle(component, fill_layers, fill_size, avoid_layers, include_layers, margin, fill_densities, fill_inverted, bbox)
    237         polygons=include_polys, bounds=bbox, dx=fill_size[0], dy=fill_size[1]
    238     )
--> 239     raster = _expand_raster(raster, distance=margin / np.array(fill_size))
    240 
    241     for i in range(np.size(raster, 0)):

~\anaconda3\lib\site-packages\gdsfactory\fill.py in _expand_raster(raster, distance)
    103     neighborhood[rr, cc] = 1
    104 
--> 105     return morphology.binary_dilation(image=raster, footprint=neighborhood)
    106 
    107 

~\anaconda3\lib\site-packages\skimage\morphology\misc.py in func_out(image, selem, *args, **kwargs)
     37         if selem is None:
     38             selem = _default_selem(image.ndim)
---> 39         return func(image, selem=selem, *args, **kwargs)
     40 
     41     return func_out

TypeError: binary_dilation() got an unexpected keyword argument 'footprint'

**Could you please help me to figure it out the reason? thanks a lot!**

**Comment by simbilod** [comment, marked as answer]
it runs for me. Is your skimage package the right version? Try uninstalling it from your environment and reinstalling gdsfactory

![image](https://user-images.githubusercontent.com/46427609/227603379-f3c92eca-e14e-4756-b1d8-a7ac02d8dae5.png)
